relayer/chains/substrate/query.go

```diff
@@ -0,0 +1,519 @@
+package substrate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	transfertypes "github.com/cosmos/ibc-go/v5/modules/apps/transfer/types"
+	clienttypes "github.com/cosmos/ibc-go/v5/modules/core/02-client/types"
+	conntypes "github.com/cosmos/ibc-go/v5/modules/core/03-connection/types"
+	chantypes "github.com/cosmos/ibc-go/v5/modules/core/04-channel/types"
+	committypes "github.com/cosmos/ibc-go/v5/modules/core/23-commitment/types"
+	ibcexported "github.com/cosmos/ibc-go/v5/modules/core/exported"
+	"github.com/cosmos/relayer/v2/relayer/provider"
+	"golang.org/x/sync/errgroup"
+)
+
+var _ provider.QueryProvider = &SubstrateProvider{}
+
+// QueryTx takes a transaction hash and returns the transaction
+func (sp *SubstrateProvider) QueryTx(ctx context.Context, hashHex string) (*provider.RelayerTxResponse, error) {
+	return &provider.RelayerTxResponse{}, errors.New(ErrTextSubstrateDoesNotHaveQueryForTransactions)
+}
+
+// QueryTxs returns an array of transactions given a tag
+func (sp *SubstrateProvider) QueryTxs(ctx context.Context, page, limit int, events []string) ([]*provider.RelayerTxResponse, error) {
+	return []*provider.RelayerTxResponse{}, errors.New(ErrTextSubstrateDoesNotHaveQueryForTransactions)
+}
+
+// QueryBalance returns the amount of coins in the relayer account
+func (sp *SubstrateProvider) QueryBalance(ctx context.Context, keyName string) (sdk.Coins, error) {
+	var (
+		addr string
+		err  error
+	)
+	if keyName == "" {
+		addr, err = sp.Address()
+	} else {
+		sp.Config.Key = keyName
+		addr, err = sp.Address()
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	return sp.QueryBalanceWithAddress(ctx, addr)
+}
+
+// QueryBalanceWithAddress returns the amount of coins in the relayer account with address as input
+// TODO add pagination support
+func (sp *SubstrateProvider) QueryBalanceWithAddress(ctx context.Context, address string) (sdk.Coins, error) {
+	// TODO: addr might need to be passed as byte not string
+	res, err := sp.RPCClient.RPC.IBC.QueryBalanceWithAddress(ctx, []byte(address))
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+// QueryUnbondingPeriod returns the unbonding period of the chain
+func (sp *SubstrateProvider) QueryUnbondingPeriod(ctx context.Context) (time.Duration, error) {
+	return 0, nil
```

Comment:
After some light Googling I see that Polkadot has an unbonding period of 28 days, my naive understanding is that all stake in Polkadot is locked up on the relay chain so does that mean the 28 day unbonding period applies in all cases no matter what Parachain may be actually connecting to IBC?

If so could this be hardcoded to 28 days?